pytorch_lightning/plugins/environments/slurm_environment.py

```diff
@@ -42,6 +43,21 @@ def detect() -> bool:
         """Returns ``True`` if the current process was launched on a SLURM cluster."""
         return "SLURM_NTASKS" in os.environ
 
+    @staticmethod
+    def job_id() -> Optional[int]:
```

Comment:
we can drastically improve code quality here in a follow up PR

Reply:
@awaelchli would you prefer introducing `job_name` as a separate staticmethod in this PR itself?

Reply:
Should it be included in this PR, or just the deprecation of job_id in Trainer?

Reply:
just the deprecation here, no further changes :) 
@ananthsub yes that sounds like a good idea to me! @AndresAlgaba interested for doing this in a follow-up PR?

Reply:
Yes, sure!